docs/Integration_Guide.md

```diff
@@ -0,0 +1,115 @@
+
+#### Entities and addresses
+
+- **Operator**: The address (whether EOA or a contract) that actaully configures and updates the protection of the contract. This can be either the protocol developer/manager, the SphereX tech team, or any other wallet that has the permission to do so.
+- **Admin**: The address (whether EOA or a contract) that controls the operator. This should be the protocol owner , and it's meant as a protection against a rogue operator.
+- **Allowed Senders** - One of our protection mechanisms is to only allow pre approved addresses to send data to the engine. So this is just a list of all the contracts that are allowed to send data to the engine.
+  - This becomes important if you deploy contracts dynamically (for example, using the factory pattern).
+
+This is a good place to emphesize and reiterate: the **Operator** and the **Admin** can be a wallet address (EOA) but cal also be a smart contract (like a DAO contract or a multisig)
+
+This also means that if indeed we use a smart contract as the operator or the admin, we will need to set the various roles accordingly, and also implement in that *managine contract* the various managment functions needed:
+- An Admin should be able to set the operator
+- An Operator has many many funcionalities (like setting the engine, adding and removing allowed senders and patterns, and even configuring the active rules) - Please talk to the SphereX team to help you understand the various roles and permissions.
+
+If the protocol has some kind of a registry or management contract, it might be a good candidate for the operator/admin roles.
+
+## Integration
+
+### Inline vs Proxy
+
+The SphereX protection has 2 ways to integrate with a given smart contract:
+1. **Inline**: The protection logic is integrated directly into the smart contract as function *decorators*, by inhereting from the contract `SphereXProtected`. This is used for immutable contracts.
+1. **Proxy**: The protection logic is integrated into a proxy contract that wraps the protected contract. It is the proxy that communicates with the engine before and after passing the calls to the implementation contract. The integration is done by replacing the existing proxy with one of the SphereX proxies (there are many variation of the *protected proxy*) or inhereting directly from `SphereXProtectedProxy`. This is used for mutalbe\upgradable contracts.
+   - Proxy protection also allows to **configure what function are protected after deployments**. So you can decide for a given function that it does not require protection, and configure the engine and proxy to ignore it.
+
+It is entirly possible (and quite common) to find both types of protection for contracts in a given protocol, as some of the underlying contracts are immutable and some are mutable.
```

Comment:
entirely

Reply:
because it's a common figure of speech

Reply:
But it's not a speech. It's a document. Not sure I would use it in this context